### ISSUE TITLE
Different training data used with SUBG and RWG methods (CivilComments)

### ISSUE THREAD

**Issue by dteney** [facebookresearch/BalancingGroups#7]
With CivilComments, the methods SUBG and RWG use a different preprocessed version of the data (fine/coarse). The paper does not mention this, and only mention the "coarse" version (with "aggregated" attributes "any"/"no active attribute").

See around L177: [https://github.com/facebookresearch/BalancingGroups/blob/main/setup_datasets.py](https://github.com/facebookresearch/BalancingGroups/blob/main/setup_datasets.py)

The issue is that the "fine" training data is substantially different, because it does not consider the "no active attribute". It only includes training examples with an active attribute. As a result the size of the training set is larger (269038) for the coarse version than the fine version (148304). The proportions of labels is also quite different (88/11 coarse as reported in the paper, 82/17 fine).

While using less training data, SUBG/RWG only use training data with an "active attribute", and this could explain why these methods are so much better in worst-group accuracy (as seen in the paper). I would suspect they are worse on test data with "no active attribute". No such data is included by the preprocessing so one cannot easily check this. Can the authors comment?

Another difference is that the training data in the fine version has examples duplicated when they had multiple attributes in the original data. Only ~72% of the training examples are actually unique. Was this intended?
